src/header/common/link.rs

```diff
@@ -0,0 +1,622 @@
+                    }
+                },
+                "title" => {
+                    // Parse target attribute: `title`.
+                    // https://tools.ietf.org/html/rfc5988#section-5.4
+                    if link_header.title.is_none() {
+                        link_header.title = match link_param_split.next() {
+                            None => return Err(::Error::Header),
+                            Some(title_str) => match verify_and_trim(title_str, '"', '"') {
+                                Err(_) => return Err(::Error::Header),
+                                Ok(title) => Some(String::from(title)),
+                            },
+                        };
+                    }
+                },
+                "title*" => {
+                    // Parse target attribute: `title*`.
+                    // https://tools.ietf.org/html/rfc5988#section-5.4
+                    if link_header.title_star.is_none() {
+                        link_header.title_star = match link_param_split.next() {
+                            None => return Err(::Error::Header),
+                            Some(title_str) => Some(String::from(title_str)),
+                        };
+                    }
+                },
+                "type" => {
+                    // Parse target attribute: `type`.
+                    // https://tools.ietf.org/html/rfc5988#section-5.4
+                    if link_header.media_type.is_none() {
+                        link_header.media_type = match link_param_split.next() {
+                            None => return Err(::Error::Header),
+                            Some(media_type_str) => match verify_and_trim(media_type_str, '"', '"') {
+                                Err(_) => return Err(::Error::Header),
+                                Ok(media_type) => match media_type.parse::<Mime>() {
+                                    Err(_) => return Err(::Error::Header),
+                                    Ok(mime_type) => Some(mime_type),
+                                },
+                            },
+
+                        };
+                    }
+                },
+                "link-extension" => {
+                    // Parse target attribute: `link-extension`.
+                    // https://tools.ietf.org/html/rfc5988#section-5.4
+                    if link_header.link_extension.is_none() {
+                        link_header.link_extension = match link_param_split.next() {
+                            None => return Err(::Error::Header),
+                            Some(link_extension_str) => Some(String::from(link_extension_str)),
+                        };
+                    }
+                },
+                _ => {
+                    return Err(::Error::Header);
+                }
+            }
+        }
+
+        Ok(link_header)
+    }
+}
+
+fn verify_and_trim(s: &str, start: char, end: char) -> ::Result<&str> {
```

Comment:
It looks like `verify_and_trim` only ever is used with ASCII characters, so it can be done more efficiently with bytes. Changing `start` and `end` to `u8`, and calls to `verify_and_trim(s, b'"', b'"')`.